MCMC/MCMC-flow/project.py

```diff
@@ -0,0 +1,137 @@
+"""Define the project's workflow logic and operation functions.
+Execute this script directly from the command line, to view your project's
+status, execute operations and submit them to a cluster. See also:
+    $ python src/project.py --help
+"""
+import sys
+import os
+
+# getting the name of the directory
+# where the this file is present.
+current = os.path.dirname(os.path.realpath(__file__))
+
+# Getting the parent directory name
+# where the current directory is present.
+parent = os.path.dirname(current)
+
+# adding the parent directory to
+# the sys.path.
+sys.path.append(parent)
+
+from flow import FlowProject, directives
+from flow.environment import DefaultSlurmEnvironment
+from simulation import Simulation
+from utils import inverse_distance_attractive, inverse_distance_repulsive, lj_energy
+
+ENERGY_FUNCS = {"lj": lj_energy,
+                "i_dist_att": inverse_distance_attractive,
+                "i_dist_rep": inverse_distance_repulsive}
+
+
+class MyProject(FlowProject):
+    pass
+
+
+class Borah(DefaultSlurmEnvironment):
+    hostname_pattern = "borah"
+    template = "borah.sh"
+
+    @classmethod
+    def add_args(cls, parser):
+        parser.add_argument(
+            "--partition",
+            default="shortgpu",
+            help="Specify the partition to submit to."
+        )
+
+
+class R2(DefaultSlurmEnvironment):
+    hostname_pattern = "r2"
+    template = "r2.sh"
+
+    @classmethod
+    def add_args(cls, parser):
+        parser.add_argument(
+            "--partition",
+            default="shortgpuq",
+            help="Specify the partition to submit to."
+        )
+
+
+class Fry(DefaultSlurmEnvironment):
+    hostname_pattern = "fry"
+    template = "fry.sh"
+
+    @classmethod
+    def add_args(cls, parser):
+        parser.add_argument(
+            "--partition",
+            default="batch",
+            help="Specify the partition to submit to."
+        )
+
+
+# Definition of project-related labels (classification)
+@MyProject.label
+def sampled(job):
+    return job.doc.get("done")
+
+
+@MyProject.label
+def initialized(job):
+    return job.isfile("trajectory.gsd")
+
+
+@directives(executable="python -u")
+@directives(ngpu=1)
```

Comment:
Looks like we'll have to remove this line. It's causing signac to only request GPU nodes.